### ISSUE TITLE
[occm] Shared loadbalancer should not be renamed.

### ISSUE THREAD

**Issue by kratisto** [kubernetes/cloud-provider-openstack#2682]
Hello, 

Is this a BUG REPORT or FEATURE REQUEST?:
/kind bug

While testing shared loadbalancers, i think i discovered a regression since this [PR](https://github.com/kubernetes/cloud-provider-openstack/pull/2552). 

A shared loadbalancer is renamed due to [this if](https://github.com/kubernetes/cloud-provider-openstack/blob/d069b1ef5d34e1400c798424f5d56b11098fb9a7/pkg/openstack/loadbalancer.go#L1694). What happens here is : 

We enter the `ensureOctaviaLoadBalancer` function as `clusterb` with a shared loadbalancer named : `kube_service_clustera_kns_fakeservice`

At the end we can see the cluster has been renamed : `kube_service_clusterb_kns_fakeservice`.

And we have the event : 
`Warning LoadBalancerRename Loadbalancer fakeloadbalancerid has a name of kube_service_clustera_kns_fakeservice with incorrect cluster-name component. Renaming it to kube_service_clusterb_kns_fakeservice.`

The tags has been changed too : 
`["kube_service_clustera_kns_fakeservice","clustera","kube_service_clusterb_kns_fakeservice"]`
become `["kube_service_clusterb_kns_fakeservice","clustera","kube_service_clusterb_kns_fakeservice"]`

I created a test that reproduce the [regression](https://gist.github.com/kratisto/b9918274314bf45582a9454dbf9d24ed). 

Didn't test if it's the case , but i suspect that if we launch a third time as `clusterc` , the shared loadbalancer will be renamed again. 
So the ownership will be for the last runner of `ensureOctaviaLoadBalancer`

**Comment by mxm-tr** [comment]
Yep, this prevents LBs from being shared between multiple clusters
